packages/web/src/pages/sign-up-page/utils/RouteContext.tsx

```diff
@@ -0,0 +1,36 @@
+import { ReactNode, createContext, useEffect, useState } from 'react'
+
+import { useHistory } from 'react-router-dom'
+
+export const RouteContext = createContext({
+  isGoBack: false
+})
+
+type RouteContextProviderProps = {
+  children: ReactNode
+}
+
+export const RouteContextProvider = (props: RouteContextProviderProps) => {
+  const { children } = props
+
+  const history = useHistory()
+  const [isGoBack, setIsGoBack] = useState(false)
+
+  useEffect(() => {
+    const unlisten = history.listen((_location, action) => {
+      if (action === 'POP') {
+        setIsGoBack(true)
+      } else {
+        setIsGoBack(false)
+      }
+    })
+
```

Comment:
Nice! Love the provider you made here ❤️ ! I'm even wondering if we may eventually want to use this elsewhere in the future 🤔 

I think what you have already looks good so imo this is perfectly fine as is, but I did find that it treats "forwards" as backs too in this case since we're not checking the to/from destinations. Probably "nice-to-have" at this point, but if you feel like messing with it I wanna see it haha 👀 

https://www.loom.com/share/c3dfb3e04ab94d4aa586e308ede36c56